migration-engine/connectors/sql-migration-connector/src/flavour/sqlite.rs

```diff
@@ -15,6 +15,10 @@ pub(crate) struct SqliteFlavour {
 
 #[async_trait::async_trait]
 impl SqlFlavour for SqliteFlavour {
+    async fn acquire_lock(&self, _connection: &Connection) -> ConnectorResult<()> {
+        Ok(())
+    }
```

Comment:
This somewhat worries me. The API promises to acquire a lock, or fail if doesn't. But instead we always succeed here even if we're never able to acquire a lock. This makes this API misbehave entirely.

Instead it seems better to:
- Either provide some way to detect whether locks can be acquired; e.g. expose a `advisory_lock_supported` method, or some other way to communicate capabilities.
- Or otherwise ensure that all databases implement advisory locking; for `sqlite` this could for example be done by creating a pid file and applying an advisory lock through [`fd_lock`](https://docs.rs/fd-lock/2.0.0/fd_lock/) [1]. This implements cross-platform advisory locks at the OS-level, and together with tokio's timeout function should be enough to emulate much of the same functionality.

[1]: `fd-lock` is a crate I've authored. It's about 100 lines total, but if there are any issues with it it should be easy enough to fix.

Reply:
The method name is a bit overpromising, but the api doc on MigrationConnector is rather clear that it's best-effort: "/// If possible on the target connector, acquire an advisory lock, so multiple instances of migrate do not run concurrently.
". I haven't paid much attention to SQLite because it is purely local and has a single concurrent writer. You're right that it could be a problem if you start several `prisma migrate deploy` concurrently on the same file, though (process a and b compute what migrations to apply, then apply the same sequence of migrations one after the other). That prompted me to do some more research, and SQLite has exclusive locks built-in via [the locking_mode pragma](https://sqlite.org/pragma.html#pragma_locking_mode). I'll just implement `acquire_lock()` that way on SQLite, commit incoming.